enhancements/proposals/cleanup_after_scaling.md

```diff
@@ -0,0 +1,140 @@
+because the cleanup part is not implemented. It’s important because upon scaling, the stored on node disk might become
+stale taking up unnecessary space. Operator should support running the cleanup to keep disk space as low as possible and
+ensure clusters are stable and reliable over time.
+
+## Motivation
+
+The motivation behind running node cleanup after scaling a cluster is to avoid the accumulation of unnecessary data on
+the node disks. When nodes are added or removed from the cluster, they gain or lose some tokens, which can result in
+files stored on the node disks still containing data associated with lost tokens. Over time, this can lead to a build-up
+of unnecessary data and cause disk space issues. By running node cleanup after scaling, these files can be cleared,
+freeing up disk space.
+
+### Goals
+
+* Running a node cleanup procedure after horizontally scaling the cluster.
+* Running a node cleanup procedure after scaling is completed, on all nodes but the last one added.
+
+### Non-Goals
+
+Running node cleanup during off-peak hours.
+
+## Proposal
+
+I propose to extend the sidecar with a new controller that will project the latest cleaned up and existing hash of
+tokens for each node as an annotation in the member service.
+In addition, a new controller in Operator will be responsible for managing Jobs that will execute a cleanup on nodes
+that require it. The trigger for the Job creation will be a mismatch between the current and latest hash. The controller
+will ensure that there will be only one cleanup Job running at the same time to prevent extraneous load on the cluster
+as cleanup might be an expensive operation.
+
+### Risks and Mitigations
+
+#### Cluster bootstrap time is increased
+
+When a cluster is created for the first time, tokens are reshuffled when new nodes are joining sequentially. This will
+cause the token hash mismatch and will require a cleanup once all nodes join the cluster. This will cause cluster
+bootstrap time to be increased. But since the time window is short, nodes shouldn’t have much data stored so cleanup
+won’t take long.
+
+#### Cleanup is run when not necessary
+
+This design doesn’t take into account whether a node received a token or lost it, it only detects ring changes and
+reacts with a cleanup trigger upon change. When a node is decommissioned, tokens are redistributed and nodes getting
+them doesn’t require a cleanup since there’s no stale data on their disks associated with these new tokens. Operator
+will trigger the cleanup anyway.
+ScyllaDB plans to automate cleanup and cover all the cases, until then we will run unnecessary cleanup.
+
+#### Cleanup is not run when necessary
+
+When keyspace RF is decreased, nodes no longer need to keep extraneous copies of the data, cleanup could free the disks.
+Approach designed here doesn't detect this case because the token ring is not changed.
```

Comment:
Similarly, cleanup is needed after changing replication strategy, e.g. from Simple to NetworkTopology as token ownership of secondary replica will change.

By the way, speaking of automation in this respect - when increasing RF, repair need to run in order to build the additional replicas.  This is probably not automated as well IIUC.

Reply:
Added mention about it.